### ISSUE TITLE
Custom Column after aggregations does not include fields in joined Saved Question

### ISSUE THREAD

**Issue by flamber** [metabase/metabase#17770]
**Describe the bug**
Questions with Custom Column after aggregations cannot be used as joining tables in nested question.
Regression since 0.38.0

**To Reproduce**
1. Custom question > Sample Dataset > Products - summarize Count by Category, and created Custom Column `1 + 1` as CC, and save question as "Q1"
![image](https://user-images.githubusercontent.com/1447303/132101728-d961b470-5aee-4b90-9613-294c8bd34422.png)
2. Custom question > Saved Questions > "Q1" - join "Q1" again (this is just for simplicity) on CC=CC
(It could also have been Products as the base table joining in "Q1")
![image](https://user-images.githubusercontent.com/1447303/132101910-8caebf55-a55c-4589-a95b-b8aa559f3def.png)
3. Visualizing will error with `Column "Question 905.count" not found;`

<details><summary>Full stacktrace</summary>

```
2021-09-04 18:45:11,663 INFO api.dataset :: Source query for this query is Card 905
2021-09-04 18:45:11,671 INFO middleware.fetch-source-query :: Fetched source query from Card 905: 
 {:source-query {:source-table 10, :aggregation [[:count]], :breakout [[:field 107 nil]]}, :expressions {:CC [:+ 1 1]}}

2021-09-04 18:45:11,681 INFO middleware.fetch-source-query :: Fetched source query from Card 905: 
 {:source-query {:source-table 10, :aggregation [[:count]], :breakout [[:field 107 nil]]}, :expressions {:CC [:+ 1 1]}}

2021-09-04 18:45:11,691 INFO middleware.fetch-source-query :: Fetched source query from Card 905: 
 {:source-query {:source-table 10, :aggregation [[:count]], :breakout [[:field 107 nil]]}, :expressions {:CC [:+ 1 1]}}

2021-09-04 18:45:11,701 INFO middleware.fetch-source-query :: Fetched source query from Card 905: 
 {:source-query {:source-table 10, :aggregation [[:count]], :breakout [[:field 107 nil]]}, :expressions {:CC [:+ 1 1]}}

2021-09-04 18:45:12,577 ERROR middleware.catch-exceptions :: Error processing query: null
{:database_id 4,
 :started_at #t "2021-09-04T18:45:11.666623+02:00[Europe/Copenhagen]",
 :via
 [{:status :failed,
   :class clojure.lang.ExceptionInfo,
   :error "Error executing query",
   :stacktrace
   ["--> driver.sql_jdbc.execute$execute_reducible_query$fn__80799.invoke(execute.clj:480)"
    "driver.sql_jdbc.execute$execute_reducible_query.invokeStatic(execute.clj:477)"
    "driver.sql_jdbc.execute$execute_reducible_query.invoke(execute.clj:463)"
    "driver.sql_jdbc.execute$execute_reducible_query.invokeStatic(execute.clj:472)"
    "driver.sql_jdbc.execute$execute_reducible_query.invoke(execute.clj:463)"
    "driver.sql_jdbc$fn__82288.invokeStatic(sql_jdbc.clj:54)"
    "driver.sql_jdbc$fn__82288.invoke(sql_jdbc.clj:52)"
    "driver.h2$fn__81040.invokeStatic(h2.clj:83)"
    "driver.h2$fn__81040.invoke(h2.clj:80)"
    "query_processor.context$executef.invokeStatic(context.clj:59)"
    "query_processor.context$executef.invoke(context.clj:48)"
    "query_processor.context.default$default_runf.invokeStatic(default.clj:68)"
    "query_processor.context.default$default_runf.invoke(default.clj:66)"
    "query_processor.context$runf.invokeStatic(context.clj:45)"
    "query_processor.context$runf.invoke(context.clj:39)"
    "query_processor.reducible$pivot.invokeStatic(reducible.clj:34)"
    "query_processor.reducible$pivot.invoke(reducible.clj:31)"
    "query_processor.middleware.mbql_to_native$mbql__GT_native$fn__47902.invoke(mbql_to_native.clj:25)"
    "query_processor.middleware.check_features$check_features$fn__47016.invoke(check_features.clj:39)"
    "query_processor.middleware.limit$limit$fn__47888.invoke(limit.clj:37)"
    "query_processor.middleware.cache$maybe_return_cached_results$fn__46468.invoke(cache.clj:204)"
    "query_processor.middleware.optimize_temporal_filters$optimize_temporal_filters$fn__48148.invoke(optimize_temporal_filters.clj:204)"
    "query_processor.middleware.validate_temporal_bucketing$validate_temporal_bucketing$fn__50080.invoke(validate_temporal_bucketing.clj:50)"
    "query_processor.middleware.auto_parse_filter_values$auto_parse_filter_values$fn__45587.invoke(auto_parse_filter_values.clj:43)"
    "query_processor.middleware.wrap_value_literals$wrap_value_literals$fn__41716.invoke(wrap_value_literals.clj:161)"
    "query_processor.middleware.annotate$add_column_info$fn__41591.invoke(annotate.clj:608)"
    "query_processor.middleware.permissions$check_query_permissions$fn__46888.invoke(permissions.clj:81)"
    "query_processor.middleware.pre_alias_aggregations$pre_alias_aggregations$fn__49009.invoke(pre_alias_aggregations.clj:40)"
    "query_processor.middleware.cumulative_aggregations$handle_cumulative_aggregations$fn__47089.invoke(cumulative_aggregations.clj:60)"
    "query_processor.middleware.resolve_joined_fields$resolve_joined_fields$fn__49306.invoke(resolve_joined_fields.clj:102)"
    "query_processor.middleware.resolve_joins$resolve_joins$fn__49619.invoke(resolve_joins.clj:171)"
    "query_processor.middleware.add_implicit_joins$add_implicit_joins$fn__45163.invoke(add_implicit_joins.clj:190)"
    "query_processor.middleware.large_int_id$convert_id_to_string$fn__47852.invoke(large_int_id.clj:59)"
    "query_processor.middleware.format_rows$format_rows$fn__47833.invoke(format_rows.clj:74)"
    "query_processor.middleware.add_default_temporal_unit$add_default_temporal_unit$fn__44457.invoke(add_default_temporal_unit.clj:23)"
    "query_processor.middleware.desugar$desugar$fn__47155.invoke(desugar.clj:21)"
    "query_processor.middleware.binning$update_binning_strategy$fn__45974.invoke(binning.clj:229)"
    "query_processor.middleware.resolve_fields$resolve_fields$fn__46691.invoke(resolve_fields.clj:34)"
    "query_processor.middleware.add_dimension_projections$add_remapping$fn__44812.invoke(add_dimension_projections.clj:314)"
    "query_processor.middleware.add_implicit_clauses$add_implicit_clauses$fn__45041.invoke(add_implicit_clauses.clj:147)"
    "query_processor.middleware.upgrade_field_literals$upgrade_field_literals$fn__50029.invoke(upgrade_field_literals.clj:40)"
    "query_processor.middleware.add_source_metadata$add_source_metadata_for_source_queries$fn__45326.invoke(add_source_metadata.clj:123)"
    "query_processor.middleware.reconcile_breakout_and_order_by_bucketing$reconcile_breakout_and_order_by_bucketing$fn__49181.invoke(reconcile_breakout_and_order_by_bucketing.clj:100)"
    "query_processor.middleware.auto_bucket_datetimes$auto_bucket_datetimes$fn__45534.invoke(auto_bucket_datetimes.clj:147)"
    "query_processor.middleware.resolve_source_table$resolve_source_tables$fn__46738.invoke(resolve_source_table.clj:45)"
    "query_processor.middleware.parameters$substitute_parameters$fn__48991.invoke(parameters.clj:111)"
    "query_processor.middleware.resolve_referenced$resolve_referenced_card_resources$fn__46790.invoke(resolve_referenced.clj:79)"
    "query_processor.middleware.expand_macros$expand_macros$fn__47539.invoke(expand_macros.clj:184)"
    "query_processor.middleware.add_timezone_info$add_timezone_info$fn__45335.invoke(add_timezone_info.clj:15)"
    "query_processor.middleware.splice_params_in_response$splice_params_in_response$fn__49982.invoke(splice_params_in_response.clj:32)"
    "query_processor.middleware.resolve_database_and_driver$resolve_database_and_driver$fn__49192$fn__49196.invoke(resolve_database_and_driver.clj:31)"
    "driver$do_with_driver.invokeStatic(driver.clj:60)"
    "driver$do_with_driver.invoke(driver.clj:56)"
    "query_processor.middleware.resolve_database_and_driver$resolve_database_and_driver$fn__49192.invoke(resolve_database_and_driver.clj:25)"
    "query_processor.middleware.fetch_source_query$resolve_card_id_source_tables$fn__47779.invoke(fetch_source_query.clj:273)"
    "query_processor.middleware.store$initialize_store$fn__49991$fn__49992.invoke(store.clj:11)"
    "query_processor.store$do_with_store.invokeStatic(store.clj:44)"
    "query_processor.store$do_with_store.invoke(store.clj:38)"
    "query_processor.middleware.store$initialize_store$fn__49991.invoke(store.clj:10)"
    "query_processor.middleware.validate$validate_query$fn__50036.invoke(validate.clj:10)"
    "query_processor.middleware.normalize_query$normalize$fn__47915.invoke(normalize_query.clj:22)"
    "query_processor.middleware.add_rows_truncated$add_rows_truncated$fn__45181.invoke(add_rows_truncated.clj:35)"
    "query_processor.middleware.results_metadata$record_and_return_metadata_BANG_$fn__49967.invoke(results_metadata.clj:147)"
    "query_processor.middleware.constraints$add_default_userland_constraints$fn__47032.invoke(constraints.clj:42)"
    "query_processor.middleware.process_userland_query$process_userland_query$fn__49078.invoke(process_userland_query.clj:134)"
    "query_processor.middleware.catch_exceptions$catch_exceptions$fn__46972.invoke(catch_exceptions.clj:173)"
    "query_processor.reducible$async_qp$qp_STAR___38255$thunk__38256.invoke(reducible.clj:103)"
    "query_processor.reducible$async_qp$qp_STAR___38255.invoke(reducible.clj:109)"
    "query_processor.reducible$sync_qp$qp_STAR___38264$fn__38267.invoke(reducible.clj:135)"
    "query_processor.reducible$sync_qp$qp_STAR___38264.invoke(reducible.clj:134)"
    "query_processor$process_userland_query.invokeStatic(query_processor.clj:241)"
    "query_processor$process_userland_query.doInvoke(query_processor.clj:237)"
    "query_processor$fn__50126$process_query_and_save_execution_BANG___50135$fn__50138.invoke(query_processor.clj:253)"
    "query_processor$fn__50126$process_query_and_save_execution_BANG___50135.invoke(query_processor.clj:245)"
    "query_processor$fn__50170$process_query_and_save_with_max_results_constraints_BANG___50179$fn__50182.invoke(query_processor.clj:265)"
    "query_processor$fn__50170$process_query_and_save_with_max_results_constraints_BANG___50179.invoke(query_processor.clj:258)"
    "api.dataset$run_query_async$fn__56417.invoke(dataset.clj:56)"
    "query_processor.streaming$streaming_response_STAR_$fn__56396$fn__56397.invoke(streaming.clj:72)"
    "query_processor.streaming$streaming_response_STAR_$fn__56396.invoke(streaming.clj:71)"
    "async.streaming_response$do_f_STAR_.invokeStatic(streaming_response.clj:65)"
    "async.streaming_response$do_f_STAR_.invoke(streaming_response.clj:63)"
    "async.streaming_response$do_f_async$fn__16088.invoke(streaming_response.clj:84)"],
   :error_type :invalid-query,
   :ex-data
   {:sql
    "-- Metabase:: userID: 1 queryType: MBQL queryHash: 082602cd46bae7457b283eb9c1f5eac9dd658572cbb5b3472d3268d90adbb995\nSELECT \"Question 905\".\"CATEGORY\" AS \"Question 905__CATEGORY\", \"source\".\"count\" AS \"count\", \"source\".\"CC\" AS \"CC\", \"Question 905\".\"count\" AS \"count_2\", \"Question 905\".\"CC\" AS \"CC_2\" FROM (SELECT \"source\".\"CATEGORY\" AS \"CATEGORY\", \"source\".\"count\" AS \"count\", (1 + 1) AS \"CC\" FROM (SELECT \"PUBLIC\".\"PRODUCTS\".\"CATEGORY\" AS \"CATEGORY\", count(*) AS \"count\" FROM \"PUBLIC\".\"PRODUCTS\" GROUP BY \"PUBLIC\".\"PRODUCTS\".\"CATEGORY\" ORDER BY \"PUBLIC\".\"PRODUCTS\".\"CATEGORY\" ASC) \"source\") \"source\" LEFT JOIN (SELECT \"source\".* FROM (SELECT \"source\".\"CATEGORY\" AS \"CATEGORY\" FROM (SELECT \"PUBLIC\".\"PRODUCTS\".\"CATEGORY\" AS \"CATEGORY\", count(*) AS \"count\" FROM \"PUBLIC\".\"PRODUCTS\" GROUP BY \"PUBLIC\".\"PRODUCTS\".\"CATEGORY\") \"source\") \"source\") \"Question 905\" ON \"source\".\"CC\" = \"Question 905\".\"CC\" LIMIT 2000",
    :params nil,
    :type :invalid-query}}],
 :state "42S22",
 :error_type :invalid-query,
 :json_query
 {:type "query",
  :query
  {:source-table "card__905",
   :joins
   [{:fields "all",
     :source-table "card__905",
     :condition
     ["="
      ["field" "CC" {:base-type "type/Integer"}]
      ["field" "CC" {:base-type "type/Integer", :join-alias "Question 905"}]],
     :alias "Question 905"}]},
  :database 4,
  :parameters [],
  :middleware {:js-int-to-string? true, :add-default-userland-constraints? true}},
 :native
 {:query
  "SELECT \"Question 905\".\"CATEGORY\" AS \"Question 905__CATEGORY\", \"source\".\"count\" AS \"count\", \"source\".\"CC\" AS \"CC\", \"Question 905\".\"count\" AS \"count_2\", \"Question 905\".\"CC\" AS \"CC_2\" FROM (SELECT \"source\".\"CATEGORY\" AS \"CATEGORY\", \"source\".\"count\" AS \"count\", (1 + 1) AS \"CC\" FROM (SELECT \"PUBLIC\".\"PRODUCTS\".\"CATEGORY\" AS \"CATEGORY\", count(*) AS \"count\" FROM \"PUBLIC\".\"PRODUCTS\" GROUP BY \"PUBLIC\".\"PRODUCTS\".\"CATEGORY\" ORDER BY \"PUBLIC\".\"PRODUCTS\".\"CATEGORY\" ASC) \"source\") \"source\" LEFT JOIN (SELECT \"source\".* FROM (SELECT \"source\".\"CATEGORY\" AS \"CATEGORY\" FROM (SELECT \"PUBLIC\".\"PRODUCTS\".\"CATEGORY\" AS \"CATEGORY\", count(*) AS \"count\" FROM \"PUBLIC\".\"PRODUCTS\" GROUP BY \"PUBLIC\".\"PRODUCTS\".\"CATEGORY\") \"source\") \"source\") \"Question 905\" ON \"source\".\"CC\" = \"Question 905\".\"CC\" LIMIT 2000",
  :params nil},
 :status :failed,
 :class org.h2.jdbc.JdbcSQLException,
 :stacktrace
 ["org.h2.message.DbException.getJdbcSQLException(DbException.java:357)"
  "org.h2.message.DbException.get(DbException.java:179)"
  "org.h2.message.DbException.get(DbException.java:155)"
  "org.h2.expression.ExpressionColumn.optimize(ExpressionColumn.java:150)"
  "org.h2.expression.Alias.optimize(Alias.java:51)"
  "org.h2.command.dml.Select.prepare(Select.java:858)"
  "org.h2.command.Parser.prepareCommand(Parser.java:283)"
  "org.h2.engine.Session.prepareLocal(Session.java:611)"
  "org.h2.engine.Session.prepareCommand(Session.java:549)"
  "org.h2.jdbc.JdbcConnection.prepareCommand(JdbcConnection.java:1247)"
  "org.h2.jdbc.JdbcStatement.executeInternal(JdbcStatement.java:217)"
  "org.h2.jdbc.JdbcStatement.execute(JdbcStatement.java:205)"
  "com.mchange.v2.c3p0.impl.NewProxyStatement.execute(NewProxyStatement.java:75)"
  "--> driver.sql_jdbc.execute$fn__80719.invokeStatic(execute.clj:344)"
  "driver.sql_jdbc.execute$fn__80719.invoke(execute.clj:342)"
  "driver.sql_jdbc.execute$execute_statement_or_prepared_statement_BANG_.invokeStatic(execute.clj:352)"
  "driver.sql_jdbc.execute$execute_statement_or_prepared_statement_BANG_.invoke(execute.clj:349)"
  "driver.sql_jdbc.execute$execute_reducible_query$fn__80799.invoke(execute.clj:478)"
  "driver.sql_jdbc.execute$execute_reducible_query.invokeStatic(execute.clj:477)"
  "driver.sql_jdbc.execute$execute_reducible_query.invoke(execute.clj:463)"
  "driver.sql_jdbc.execute$execute_reducible_query.invokeStatic(execute.clj:472)"
  "driver.sql_jdbc.execute$execute_reducible_query.invoke(execute.clj:463)"
  "driver.sql_jdbc$fn__82288.invokeStatic(sql_jdbc.clj:54)"
  "driver.sql_jdbc$fn__82288.invoke(sql_jdbc.clj:52)"
  "driver.h2$fn__81040.invokeStatic(h2.clj:83)"
  "driver.h2$fn__81040.invoke(h2.clj:80)"
  "query_processor.context$executef.invokeStatic(context.clj:59)"
  "query_processor.context$executef.invoke(context.clj:48)"
  "query_processor.context.default$default_runf.invokeStatic(default.clj:68)"
  "query_processor.context.default$default_runf.invoke(default.clj:66)"
  "query_processor.context$runf.invokeStatic(context.clj:45)"
  "query_processor.context$runf.invoke(context.clj:39)"
  "query_processor.reducible$pivot.invokeStatic(reducible.clj:34)"
  "query_processor.reducible$pivot.invoke(reducible.clj:31)"
  "query_processor.middleware.mbql_to_native$mbql__GT_native$fn__47902.invoke(mbql_to_native.clj:25)"
  "query_processor.middleware.check_features$check_features$fn__47016.invoke(check_features.clj:39)"
  "query_processor.middleware.limit$limit$fn__47888.invoke(limit.clj:37)"
  "query_processor.middleware.cache$maybe_return_cached_results$fn__46468.invoke(cache.clj:204)"
  "query_processor.middleware.optimize_temporal_filters$optimize_temporal_filters$fn__48148.invoke(optimize_temporal_filters.clj:204)"
  "query_processor.middleware.validate_temporal_bucketing$validate_temporal_bucketing$fn__50080.invoke(validate_temporal_bucketing.clj:50)"
  "query_processor.middleware.auto_parse_filter_values$auto_parse_filter_values$fn__45587.invoke(auto_parse_filter_values.clj:43)"
  "query_processor.middleware.wrap_value_literals$wrap_value_literals$fn__41716.invoke(wrap_value_literals.clj:161)"
  "query_processor.middleware.annotate$add_column_info$fn__41591.invoke(annotate.clj:608)"
  "query_processor.middleware.permissions$check_query_permissions$fn__46888.invoke(permissions.clj:81)"
  "query_processor.middleware.pre_alias_aggregations$pre_alias_aggregations$fn__49009.invoke(pre_alias_aggregations.clj:40)"
  "query_processor.middleware.cumulative_aggregations$handle_cumulative_aggregations$fn__47089.invoke(cumulative_aggregations.clj:60)"
  "query_processor.middleware.resolve_joined_fields$resolve_joined_fields$fn__49306.invoke(resolve_joined_fields.clj:102)"
  "query_processor.middleware.resolve_joins$resolve_joins$fn__49619.invoke(resolve_joins.clj:171)"
  "query_processor.middleware.add_implicit_joins$add_implicit_joins$fn__45163.invoke(add_implicit_joins.clj:190)"
  "query_processor.middleware.large_int_id$convert_id_to_string$fn__47852.invoke(large_int_id.clj:59)"
  "query_processor.middleware.format_rows$format_rows$fn__47833.invoke(format_rows.clj:74)"
  "query_processor.middleware.add_default_temporal_unit$add_default_temporal_unit$fn__44457.invoke(add_default_temporal_unit.clj:23)"
  "query_processor.middleware.desugar$desugar$fn__47155.invoke(desugar.clj:21)"
  "query_processor.middleware.binning$update_binning_strategy$fn__45974.invoke(binning.clj:229)"
  "query_processor.middleware.resolve_fields$resolve_fields$fn__46691.invoke(resolve_fields.clj:34)"
  "query_processor.middleware.add_dimension_projections$add_remapping$fn__44812.invoke(add_dimension_projections.clj:314)"
  "query_processor.middleware.add_implicit_clauses$add_implicit_clauses$fn__45041.invoke(add_implicit_clauses.clj:147)"
  "query_processor.middleware.upgrade_field_literals$upgrade_field_literals$fn__50029.invoke(upgrade_field_literals.clj:40)"
  "query_processor.middleware.add_source_metadata$add_source_metadata_for_source_queries$fn__45326.invoke(add_source_metadata.clj:123)"
  "query_processor.middleware.reconcile_breakout_and_order_by_bucketing$reconcile_breakout_and_order_by_bucketing$fn__49181.invoke(reconcile_breakout_and_order_by_bucketing.clj:100)"
  "query_processor.middleware.auto_bucket_datetimes$auto_bucket_datetimes$fn__45534.invoke(auto_bucket_datetimes.clj:147)"
  "query_processor.middleware.resolve_source_table$resolve_source_tables$fn__46738.invoke(resolve_source_table.clj:45)"
  "query_processor.middleware.parameters$substitute_parameters$fn__48991.invoke(parameters.clj:111)"
  "query_processor.middleware.resolve_referenced$resolve_referenced_card_resources$fn__46790.invoke(resolve_referenced.clj:79)"
  "query_processor.middleware.expand_macros$expand_macros$fn__47539.invoke(expand_macros.clj:184)"
  "query_processor.middleware.add_timezone_info$add_timezone_info$fn__45335.invoke(add_timezone_info.clj:15)"
  "query_processor.middleware.splice_params_in_response$splice_params_in_response$fn__49982.invoke(splice_params_in_response.clj:32)"
  "query_processor.middleware.resolve_database_and_driver$resolve_database_and_driver$fn__49192$fn__49196.invoke(resolve_database_and_driver.clj:31)"
  "driver$do_with_driver.invokeStatic(driver.clj:60)"
  "driver$do_with_driver.invoke(driver.clj:56)"
  "query_processor.middleware.resolve_database_and_driver$resolve_database_and_driver$fn__49192.invoke(resolve_database_and_driver.clj:25)"
  "query_processor.middleware.fetch_source_query$resolve_card_id_source_tables$fn__47779.invoke(fetch_source_query.clj:273)"
  "query_processor.middleware.store$initialize_store$fn__49991$fn__49992.invoke(store.clj:11)"
  "query_processor.store$do_with_store.invokeStatic(store.clj:44)"
  "query_processor.store$do_with_store.invoke(store.clj:38)"
  "query_processor.middleware.store$initialize_store$fn__49991.invoke(store.clj:10)"
  "query_processor.middleware.validate$validate_query$fn__50036.invoke(validate.clj:10)"
  "query_processor.middleware.normalize_query$normalize$fn__47915.invoke(normalize_query.clj:22)"
  "query_processor.middleware.add_rows_truncated$add_rows_truncated$fn__45181.invoke(add_rows_truncated.clj:35)"
  "query_processor.middleware.results_metadata$record_and_return_metadata_BANG_$fn__49967.invoke(results_metadata.clj:147)"
  "query_processor.middleware.constraints$add_default_userland_constraints$fn__47032.invoke(constraints.clj:42)"
  "query_processor.middleware.process_userland_query$process_userland_query$fn__49078.invoke(process_userland_query.clj:134)"
  "query_processor.middleware.catch_exceptions$catch_exceptions$fn__46972.invoke(catch_exceptions.clj:173)"
  "query_processor.reducible$async_qp$qp_STAR___38255$thunk__38256.invoke(reducible.clj:103)"
  "query_processor.reducible$async_qp$qp_STAR___38255.invoke(reducible.clj:109)"
  "query_processor.reducible$sync_qp$qp_STAR___38264$fn__38267.invoke(reducible.clj:135)"
  "query_processor.reducible$sync_qp$qp_STAR___38264.invoke(reducible.clj:134)"
  "query_processor$process_userland_query.invokeStatic(query_processor.clj:241)"
  "query_processor$process_userland_query.doInvoke(query_processor.clj:237)"
  "query_processor$fn__50126$process_query_and_save_execution_BANG___50135$fn__50138.invoke(query_processor.clj:253)"
  "query_processor$fn__50126$process_query_and_save_execution_BANG___50135.invoke(query_processor.clj:245)"
  "query_processor$fn__50170$process_query_and_save_with_max_results_constraints_BANG___50179$fn__50182.invoke(query_processor.clj:265)"
  "query_processor$fn__50170$process_query_and_save_with_max_results_constraints_BANG___50179.invoke(query_processor.clj:258)"
  "api.dataset$run_query_async$fn__56417.invoke(dataset.clj:56)"
  "query_processor.streaming$streaming_response_STAR_$fn__56396$fn__56397.invoke(streaming.clj:72)"
  "query_processor.streaming$streaming_response_STAR_$fn__56396.invoke(streaming.clj:71)"
  "async.streaming_response$do_f_STAR_.invokeStatic(streaming_response.clj:65)"
  "async.streaming_response$do_f_STAR_.invoke(streaming_response.clj:63)"
  "async.streaming_response$do_f_async$fn__16088.invoke(streaming_response.clj:84)"],
 :context :ad-hoc,
 :error
 "Column \"Question 905.count\" not found; SQL statement:\n-- Metabase:: userID: 1 queryType: MBQL queryHash: 082602cd46bae7457b283eb9c1f5eac9dd658572cbb5b3472d3268d90adbb995\nSELECT \"Question 905\".\"CATEGORY\" AS \"Question 905__CATEGORY\", \"source\".\"count\" AS \"count\", \"source\".\"CC\" AS \"CC\", \"Question 905\".\"count\" AS \"count_2\", \"Question 905\".\"CC\" AS \"CC_2\" FROM (SELECT \"source\".\"CATEGORY\" AS \"CATEGORY\", \"source\".\"count\" AS \"count\", (1 + 1) AS \"CC\" FROM (SELECT \"PUBLIC\".\"PRODUCTS\".\"CATEGORY\" AS \"CATEGORY\", count(*) AS \"count\" FROM \"PUBLIC\".\"PRODUCTS\" GROUP BY \"PUBLIC\".\"PRODUCTS\".\"CATEGORY\" ORDER BY \"PUBLIC\".\"PRODUCTS\".\"CATEGORY\" ASC) \"source\") \"source\" LEFT JOIN (SELECT \"source\".* FROM (SELECT \"source\".\"CATEGORY\" AS \"CATEGORY\" FROM (SELECT \"PUBLIC\".\"PRODUCTS\".\"CATEGORY\" AS \"CATEGORY\", count(*) AS \"count\" FROM \"PUBLIC\".\"PRODUCTS\" GROUP BY \"PUBLIC\".\"PRODUCTS\".\"CATEGORY\") \"source\") \"source\") \"Question 905\" ON \"source\".\"CC\" = \"Question 905\".\"CC\" LIMIT 2000 [42122-197]",
 :row_count 0,
 :running_time 0,
 :preprocessed
 {:type :query,
  :query
  {:source-metadata
   [{:description "The type of product, valid values include: Doohicky, Gadget, Gizmo and Widget",
     :semantic_type :type/Category,
     :coercion_strategy nil,
     :name "CATEGORY",
     :field_ref [:field 107 {:join-alias "Question 905"}],
     :effective_type :type/Text,
     :id 107,
     :display_name "Category",
     :fingerprint
     {:global {:distinct-count 4, :nil% 0},
      :type {:type/Text {:percent-json 0, :percent-url 0, :percent-email 0, :percent-state 0, :average-length 6.375}}},
     :base_type :type/Text}
    {:name "count",
     :display_name "Count",
     :base_type :type/BigInteger,
     :semantic_type :type/Quantity,
     :field_ref [:field "count" {:base-type :type/BigInteger}],
     :fingerprint
     {:global {:distinct-count 4, :nil% 0},
      :type {:type/Number {:min 42, :q1 46.5, :q3 53.5, :max 54, :sd 5.477225575051661, :avg 50}}}}
    {:name "CC",
     :display_name "CC",
     :base_type :type/Integer,
     :semantic_type :type/Quantity,
     :field_ref [:expression "CC"],
     :fingerprint
     {:global {:distinct-count 1, :nil% 0}, :type {:type/Number {:min 2, :q1 2, :q3 2, :max 2, :sd 0, :avg 2}}}}],
   :fields
   [[:field 107 {:join-alias "Question 905"}]
    [:field "count" {:base-type :type/BigInteger}]
    [:field "CC" {:base-type :type/Integer}]
    [:field "count" {:base-type :type/BigInteger, :join-alias "Question 905"}]
    [:field "CC" {:base-type :type/Integer, :join-alias "Question 905"}]],
   :joins
   [{:strategy :left-join,
     :condition
     [:=
      [:field "CC" {:base-type :type/Integer}]
      [:field "CC" {:base-type :type/Integer, :join-alias "Question 905"}]],
     :alias "Question 905",
     :source-query
     {:expressions {:CC [:+ 1 1]},
      :source-metadata
      [{:semantic_type :type/Category,
        :table_id 10,
        :coercion_strategy nil,
        :name "CATEGORY",
        :settings nil,
        :field_ref [:field 107 nil],
        :effective_type :type/Text,
        :parent_id nil,
        :id 107,
        :display_name "Category",
        :fingerprint
        {:global {:distinct-count 4, :nil% 0.0},
         :type
         {:type/Text
          {:percent-json 0.0, :percent-url 0.0, :percent-email 0.0, :percent-state 0.0, :average-length 6.375}}},
        :base_type :type/Text}
       {:name "count",
        :display_name "Count",
        :base_type :type/BigInteger,
        :semantic_type :type/Quantity,
        :field_ref [:aggregation 0]}],
      :source-query
      {:source-table 10,
       :aggregation [[:aggregation-options [:count] {:name "count"}]],
       :breakout [[:field 107 nil]]}},
     :source-metadata
     [{:description "The type of product, valid values include: Doohicky, Gadget, Gizmo and Widget",
       :semantic_type :type/Category,
       :coercion_strategy nil,
       :name "CATEGORY",
       :field_ref [:field 107 nil],
       :effective_type :type/Text,
       :id 107,
       :display_name "Category",
       :fingerprint
       {:global {:distinct-count 4, :nil% 0},
        :type
        {:type/Text {:percent-json 0, :percent-url 0, :percent-email 0, :percent-state 0, :average-length 6.375}}},
       :base_type :type/Text}
      {:name "count",
       :display_name "Count",
       :base_type :type/BigInteger,
       :semantic_type :type/Quantity,
       :field_ref [:field "count" {:base-type :type/BigInteger}],
       :fingerprint
       {:global {:distinct-count 4, :nil% 0},
        :type {:type/Number {:min 42, :q1 46.5, :q3 53.5, :max 54, :sd 5.477225575051661, :avg 50}}}}
      {:name "CC",
       :display_name "CC",
       :base_type :type/Integer,
       :semantic_type :type/Quantity,
       :field_ref [:expression "CC"],
       :fingerprint
       {:global {:distinct-count 1, :nil% 0}, :type {:type/Number {:min 2, :q1 2, :q3 2, :max 2, :sd 0, :avg 2}}}}]}],
   :source-query
   {:expressions {:CC [:+ 1 1]},
    :source-metadata
    [{:semantic_type :type/Category,
      :table_id 10,
      :coercion_strategy nil,
      :name "CATEGORY",
      :settings nil,
      :field_ref [:field 107 nil],
      :effective_type :type/Text,
      :parent_id nil,
      :id 107,
      :display_name "Category",
      :fingerprint
      {:global {:distinct-count 4, :nil% 0.0},
       :type
       {:type/Text
        {:percent-json 0.0, :percent-url 0.0, :percent-email 0.0, :percent-state 0.0, :average-length 6.375}}},
      :base_type :type/Text}
     {:name "count",
      :display_name "Count",
      :base_type :type/BigInteger,
      :semantic_type :type/Quantity,
      :field_ref [:aggregation 0]}],
    :fields [[:field 107 nil] [:field "count" {:base-type :type/BigInteger}] [:expression "CC"]],
    :source-query
    {:source-table 10,
     :aggregation [[:aggregation-options [:count] {:name "count"}]],
     :breakout [[:field 107 nil]],
     :order-by [[:asc [:field 107 nil]]]}},
   :limit 2000},
  :database 4,
  :middleware {:js-int-to-string? true, :add-default-userland-constraints? true},
  :info
  {:executed-by 1,
   :context :ad-hoc,
   :card-id 905,
   :nested? true,
   :query-hash
   [8, 38, 2, -51, 70, -70, -25, 69, 123, 40, 62, -71, -63, -11, -22, -55, -35, 101, -123, 114, -53, -75, -77, 71, 45,
    50, 104, -39, 10, -37, -71, -107]},
  :constraints {:max-results 10000, :max-results-bare-rows 2000}},
 :data {:rows [], :cols []}}

2021-09-04 18:45:12,583 DEBUG middleware.log :: POST /api/dataset 202 [ASYNC: completed] 921.6 ms (22 DB calls) App DB connections: 0/10 Jetty threads: 3/50 (1 idle, 0 queued) (72 total active threads) Queries in flight: 0 (0 queued); h2 DB 4 connections: 0/1 (0 threads blocked)
```

</details>

Notice the missing metric ("count") and Custom Column ("CC") from the joined in question ("Question 905").
```SQL
SELECT
  "Question 905"."CATEGORY" AS "Question 905__CATEGORY",
  "source"."count" AS "count",
  "source"."CC" AS "CC",
  "Question 905"."count" AS "count_2",
  "Question 905"."CC" AS "CC_2" 
FROM
  (
    SELECT
      "source"."CATEGORY" AS "CATEGORY",
      "source"."count" AS "count",
      (
        1 + 1
      )
      AS "CC" 
    FROM
      (
        SELECT
          "PUBLIC"."PRODUCTS"."CATEGORY" AS "CATEGORY",
          count(*) AS "count" 
        FROM
          "PUBLIC"."PRODUCTS" 
        GROUP BY
          "PUBLIC"."PRODUCTS"."CATEGORY" 
        ORDER BY
          "PUBLIC"."PRODUCTS"."CATEGORY" ASC
      )
      "source"
  )
  "source" 
  LEFT JOIN
    (
      SELECT
        "source".* 
      FROM
        (
          SELECT
            "source"."CATEGORY" AS "CATEGORY"
          FROM
            (
              SELECT
                "PUBLIC"."PRODUCTS"."CATEGORY" AS "CATEGORY",
                count(*) AS "count" 
              FROM
                "PUBLIC"."PRODUCTS" 
              GROUP BY
                "PUBLIC"."PRODUCTS"."CATEGORY"
            )
            "source"
        )
        "source"
    )
    "Question 905" 
    ON "source"."CC" = "Question 905"."CC"
```

**Information about your Metabase Installation:**
Tested 0.37.8 thru 0.40.3.1 and master - regression since 0.38.0

**Comment by nemanjaglumac** [comment]
Won't test this with Cypress.
It's a backend test scenario.

**Comment by camsaul** [comment]
Sounds similar or duplicate (?) to #15578

**Comment by flamber** [comment]
Not a duplicate. This is a regression since 0.38, whereas #15578 has been a problem since at least 0.36.

**Comment by camsaul** [comment]
This will be fixed by #19384

**Comment by camsaul** [comment]
Fixed by #19384
